backend/open_webui/models/prompts.py

```
@@ -103,32 +104,246 @@ def get_prompt_by_command(self, command: str) -> Optional[PromptModel]:
-                            "user": user.model_dump() if user else None,
+                            "user": user_data,
                         }
                     )
                 )
 
             return prompts
 
-    def get_prompts_by_user_id(
-        self, user_id: str, permission: str = "write"
+    def get_prompts_paginated(
+        self, page: int = 1, limit: int = 20, search: str = None
```

Ruff (0.12.2)

PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)